source/extensions/tracers/skywalking/tracer.cc

```diff
@@ -0,0 +1,82 @@
+#include "extensions/tracers/skywalking/tracer.h"
+
+#include <chrono>
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace SkyWalking {
+
+constexpr char StatusCodeTag[] = "status_code";
+constexpr char UrlTag[] = "url";
+
+namespace {
+
+uint64_t getTimestamp(SystemTime time) {
+  return std::chrono::duration_cast<std::chrono::milliseconds>(time.time_since_epoch()).count();
+}
+
+} // namespace
+
+Tracing::SpanPtr Tracer::startSpan(const Tracing::Config&, SystemTime start_time,
+                                   const std::string& operation,
+                                   SegmentContextSharedPtr segment_context, Span* parent) {
+  SpanStore* span_store = segment_context->createSpanStore(parent ? parent->spanStore() : nullptr);
+
+  span_store->setStartTime(getTimestamp(start_time));
+
+  span_store->setOperation(operation);
+
+  return std::make_unique<Span>(std::move(segment_context), span_store, *this);
+}
+
+void Span::setOperation(absl::string_view operation) {
+  span_store_->setOperation(std::string(operation));
+}
+
+void Span::setTag(absl::string_view name, absl::string_view value) {
+  if (name == Tracing::Tags::get().HttpUrl) {
+    span_store_->addTag(UrlTag, value);
+    return;
+  }
+
+  if (name == Tracing::Tags::get().HttpStatusCode) {
+    span_store_->addTag(StatusCodeTag, value);
+    return;
+  }
+
+  if (name == Tracing::Tags::get().Error) {
+    span_store_->setAsError(value == Tracing::Tags::get().True);
+  }
+
+  span_store_->addTag(name, value);
+}
+
+// Logs in the SkyWalking format are temporarily unsupported.
+void Span::log(SystemTime, const std::string&) {}
```

Comment:
`NOT_IMPLEMENTED_GCOVR_EXCL_LINE`, or is this intentionally skipped?

Reply:
@lizan Yep, It is intentionally skipped.